Canceling TSV import throws unhandled exception

When a user has a table open and starts to select a file for TSV import, if the operation is canceled at the select file dialog an unhandled exception is thrown.

```
Exception in thread "JavaFX Application Thread" java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1762)
	at javafx.fxml.FXMLLoader$ControllerMethodEventHandler.handle(FXMLLoader.java:1645)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.control.MenuItem.fire(MenuItem.java:462)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.doSelect(ContextMenuContent.java:1364)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer.lambda$createChildren$324(ContextMenuContent.java:1317)
	at com.sun.javafx.scene.control.skin.ContextMenuContent$MenuItemContainer$$Lambda$282/1084824391.handle(Unknown Source)
	at com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:218)
	at com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:238)
	at com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:191)
	at com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
	at com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
	at com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
	at com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
	at javafx.event.Event.fireEvent(Event.java:198)
	at javafx.scene.Scene$MouseHandler.process(Scene.java:3724)
	at javafx.scene.Scene$MouseHandler.access$1500(Scene.java:3452)
	at javafx.scene.Scene.impl_processMouseEvent(Scene.java:1728)
	at javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2461)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:348)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(GlassViewEventHandler.java:273)
	at java.security.AccessController.doPrivileged(Native Method)
	at com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:382)
	at com.sun.glass.ui.View.handleMouseEvent(View.java:553)
	at com.sun.glass.ui.View.notifyMouse(View.java:925)
	at com.sun.glass.ui.gtk.GtkApplication._runLoop(Native Method)
	at com.sun.glass.ui.gtk.GtkApplication.lambda$null$45(GtkApplication.java:126)
	at com.sun.glass.ui.gtk.GtkApplication$$Lambda$43/584634336.run(Unknown Source)
	at java.lang.Thread.run(Thread.java:745)
Caused by: java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at sun.reflect.misc.Trampoline.invoke(MethodUtil.java:71)
	at sun.reflect.GeneratedMethodAccessor1.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:483)
	at sun.reflect.misc.MethodUtil.invoke(MethodUtil.java:275)
	at javafx.fxml.FXMLLoader$MethodHandler.invoke(FXMLLoader.java:1759)
	... 43 more
Caused by: java.lang.NullPointerException
	at org.cirdles.topsoil.app.TopsoilMainWindow.importFromFile(TopsoilMainWindow.java:299)
	... 53 more
```

--- a/src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java
+++ b/src/main/java/org/cirdles/topsoil/app/TopsoilMainWindow.java
@@ -15,6 +15,7 @@
  */
 package org.cirdles.topsoil.app;
 
+import java.io.File;
 import java.io.IOException;
 import java.net.URI;
 import java.net.URISyntaxException;
@@ -118,14 +119,7 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
     }
 
     Optional<TSVTable> getCurrentTable() {
-        Optional<TSVTable> result = Optional.empty();
-        Tab currentTab = dataTableTabPane.getSelectionModel().getSelectedItem();
-
-        if (currentTab.getContent() instanceof TSVTable) {
-            result = Optional.of((TSVTable) currentTab.getContent());
-        }
-
-        return result;
+        return getCurrentTab().map(tab -> (TSVTable) tab.getContent());
     }
 
     @FXML
@@ -164,6 +158,7 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
                     datasetName + "__open____headers__.tsv");
 
             getCurrentTable().ifPresent(table -> table.saveToPath(datasetPath));
+            getCurrentTab().ifPresent(tab -> tab.setText(datasetName));
         });
 
         // reload
@@ -239,8 +234,9 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
         }
     }
 
-    Tab getCurrentTab() {
-        return dataTableTabPane.getSelectionModel().getSelectedItem();
+    Optional<Tab> getCurrentTab() {
+        return Optional.ofNullable(
+                dataTableTabPane.getSelectionModel().getSelectedItem());
     }
 
     void setWindowTitle(String title) {
@@ -279,14 +275,12 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
         sceneProperty().addListener(sceneListener);
     }
 
-    @FXML
-    void importFromFile() {
-        TSVTable dataTable = getCurrentTable().get();
+    public TSVTable createTable() {
+        return (TSVTable) createTab().getContent();
+    }
 
-        FileChooser tsvChooser = new FileChooser();
-        tsvChooser.setInitialDirectory(Topsoil.USER_HOME.toFile());
-        tsvChooser.setSelectedExtensionFilter(new FileChooser.ExtensionFilter("Table Files", "TSV"));
-        Path filePath = tsvChooser.showOpenDialog(getScene().getWindow()).toPath();
+    void importFromFile(Path filePath) {
+        TSVTable dataTable = getCurrentTable().orElseGet(this::createTable);
 
         // JFB for now, assume error chart is only chart style
         dataTable.setRequiredColumnCount(ERROR_CHART_REQUIRED_COL_COUNT);
@@ -309,7 +303,19 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
             }
         });
     }
+    
+    @FXML
+    void importFromFile() {
+        FileChooser tsvChooser = new FileChooser();
+        tsvChooser.setInitialDirectory(Topsoil.USER_HOME.toFile());
+        tsvChooser.setSelectedExtensionFilter(
+                new FileChooser.ExtensionFilter("Table Files", "TSV"));
 
+        Optional.ofNullable(tsvChooser.showOpenDialog(getScene().getWindow()))
+                .map(File::toPath)
+                .ifPresent(this::importFromFile);
+    }
+    
     void initializeAndShow(Chart chart, Dataset dataset) {
         new VariableBindingDialog(dataset, chart).showAndWait()
                 .ifPresent(variableContext -> {
@@ -347,7 +353,7 @@ public class TopsoilMainWindow extends CustomVBox implements Initializable {
                     new JavaScriptChart(javascriptPath), dataset);
         });
     }
-
+    
     @FXML
     void createScatterplot() {
         try {

--- a/src/test/java/org/cirdles/topsoil/app/TSVTableTest.java
+++ b/src/test/java/org/cirdles/topsoil/app/TSVTableTest.java
@@ -24,7 +24,6 @@ import java.util.logging.Logger;
 import javafx.scene.Parent;
 import javafx.scene.Scene;
 import javafx.stage.Stage;
-import org.junit.Assert;
 import org.junit.Test;
 import org.testfx.framework.junit.ApplicationTest;
 

--- a/src/test/java/org/cirdles/topsoil/app/TopsoilMainWindowTest.java
+++ b/src/test/java/org/cirdles/topsoil/app/TopsoilMainWindowTest.java
@@ -16,9 +16,11 @@
 package org.cirdles.topsoil.app;
 
 import javafx.scene.Scene;
+import javafx.scene.control.TabPane;
 import javafx.stage.Stage;
+import org.junit.Test;
+import static org.testfx.api.FxAssert.verifyThat;
 import org.testfx.framework.junit.ApplicationTest;
-
 /**
  *
  * @author John Zeringue
@@ -31,5 +33,9 @@ public class TopsoilMainWindowTest extends ApplicationTest {
         stage.setScene(scene);
         stage.show();
     }
-    
+
+    @Test
+    public void testCancelImportFile() {
+        clickOn("File").clickOn("Import from TSV").closeCurrentWindow();
+    }
 }

